src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -872,15 +889,54 @@ private IEnumerable<CompletionItem> GetArrayItemCompletions(SemanticModel model,
 
         private IEnumerable<CompletionItem> GetFunctionParamCompletions(SemanticModel model, BicepCompletionContext context)
         {
-            if (!context.Kind.HasFlag(BicepCompletionContextKind.FunctionArgument) ||
-                context.FunctionArgument is null ||
-                model.GetSymbolInfo(context.FunctionArgument.Function) is not FunctionSymbol functionSymbol)
+            if (context.FunctionArgument is not { } functionArgument
+                || model.GetSymbolInfo(functionArgument.Function) is not FunctionSymbol functionSymbol)
             {
                 return Enumerable.Empty<CompletionItem>();
             }
 
             var argType = functionSymbol.GetDeclaredArgumentType(context.FunctionArgument.ArgumentIndex);
 
+            //functionArgument flag indicates that we are about to type a function argument. But path completions should be also shown when on a argument value
+            if (functionArgument.ArgumentNodes.IsEmpty
+                || !(functionArgument.ArgumentNodes[0] is StringSyntax stringSyntax && stringSyntax.IsInterpolated())) //we discard path completions when string is interpolated (although it shouldn't). here are too many cases to handle and this situation should not happen
+            {
```

Comment:
Here's my suggestion for building on top of your work to make this functionality type-based (not dependent on being part of a function argument): https://github.com/Azure/bicep/commit/3df171c77b160b6d39f556cbd91b4a2bc34780f3

Reply:
changed as you suggested